File: app/src/commonMain/kotlin/com/crosspaste/paste/plugin/process/PasteProcessPlugin.kt
package com.crosspaste.paste.plugin.process

import com.crosspaste.paste.item.PasteCoordinate
import com.crosspaste.paste.item.PasteItem

interface PasteProcessPlugin {

    fun process(
        pasteCoordinate: PasteCoordinate,
        pasteItems: List<PasteItem>,
        source: String?,
    ): List<PasteItem>
}

File: web/src/shared/paste/plugin/paste-process-plugin.ts
import type { PasteItem } from "@/shared/models/paste-item";

export interface TypedItem {
  pasteType: number;
  item: PasteItem;
}

export interface PasteProcessContext {
  hashText: (s: string) => string;
}

export interface PasteProcessPlugin {
  process(items: TypedItem[], context: PasteProcessContext): TypedItem[];
}
